src/ralph/backends/data/es.py

```diff
@@ -0,0 +1,320 @@
+        self.default_index = self.settings.DEFAULT_INDEX
+        self.locale_encoding = self.settings.LOCALE_ENCODING
+        self._client = None
+
+    @property
+    def client(self):
+        """Creates an Elasticsearch client if it doesn't exists."""
+        if not self._client:
+            self._client = Elasticsearch(
+                self.settings.HOSTS, **self.settings.CLIENT_OPTIONS.dict()
+            )
+        return self._client
+
+    def status(self) -> DataBackendStatus:
+        """Checks Elasticsearch cluster connection and status."""
+        try:
+            self.client.info()
+        except ESConnectionError as error:
+            logger.error("Failed to connect to Elasticsearch: %s", error)
+            return DataBackendStatus.AWAY
+
+        cluster_status = self.client.cat.health()
+        if "green" not in cluster_status:
+            logger.error("Cluster status is not green: %s", cluster_status)
+            return DataBackendStatus.ERROR
+
+        return DataBackendStatus.OK
+
+    def list(
+        self, target: str = None, details: bool = False, new: bool = False
+    ) -> Iterator[Union[str, dict]]:
+        """Lists available Elasticsearch indices, data streams and aliases.
+
+        Args:
+            target (str or None): The comma-separated list of data streams, indices,
+                and aliases to limit the request. Supports wildcards (*).
+                If target is `None`, lists all available indices, data streams and
+                    aliases. Equivalent to (`target` = "*").
+            details (bool): Get detailed informations instead of just names.
+            new (bool): Ignored.
+
+        Yields:
+            str: The next index, data stream or alias name. (If `details` is False).
+            dict: The next index, data stream or alias details. (If `details` is True).
+
+        Raises:
+            BackendException: If a failure during indices retrieval occurs.
+        """
+        target = target if target else "*"
+        try:
+            indices = self.client.indices.get(index=target)
+        except (ApiError, ESConnectionError) as error:
+            msg = "Failed to read indices: %s"
+            logger.error(msg, error)
+            raise BackendException(msg % error) from error
+
+        if new:
+            logger.warning("The `new` argument is ignored")
+
+        if details:
+            for index, value in indices.items():
+                yield {index: value}
+
+            return
+
+        for index in indices:
+            yield index
+
+    @enforce_query_checks
+    def read(
+        self,
+        *,
+        query: Union[str, ESQuery] = None,
+        target: str = None,
+        chunk_size: Union[None, int] = 500,
+        raw_output: bool = False,
+        ignore_errors: bool = False,
+    ) -> Iterator[Union[bytes, dict]]:
+        """Reads documents matching the query in the target index and yields them.
```

Comment:
```suggestion
        """Read documents matching the query in the target index and yield them.
```

Reply:
Good point) It seems that in most of ralph's modules, we currently use the descriptive docstring style.
However, in other parts we do use the imperative style.

The [google style guide](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) is not very strict on this part:

> The docstring may be descriptive-style ("""Fetches rows from a Bigtable.""") or imperative-style ("""Fetch rows from a Bigtable."""), but the style should be consistent within a file.

I like the idea to follow a single docstring-style across the project, was a decision already made about that?